internal/pkg/manifest/svc.go

```diff
@@ -570,3 +575,79 @@ func sortExposedPorts(exposedPorts []ExposedPort) []ExposedPort {
 	})
 	return exposedPorts
 }
+
+// HTTPLoadBalancerTarget returns target container and target port for the ALB configuration.
+func (s *LoadBalancedWebService) HTTPLoadBalancerTarget(exposedPorts []ExposedPort, rrTargetContainer *string, rrTargetPort *uint16) (targetContainer *string, targetPort *string) {
+	// Route load balancer traffic to main container by default.
+	targetContainer = s.Name
+	targetPort = aws.String(s.ContainerPort())
+
+	rrTarget := rrTargetContainer
+	if rrTarget != nil && *rrTarget != *targetContainer {
+		targetContainer = rrTarget
+		targetPort = s.Sidecars[aws.StringValue(targetContainer)].Port
+	}
+	// Route load balancer traffic to the target_port if mentioned.
+	if container, port := httpLoadBalancerTarget(exposedPorts, rrTargetPort); port != nil {
+		targetPort = port
+	} else if container != nil {
+		targetContainer = container
+	}
+	return
+}
+
+// HTTPLoadBalancerTarget returns target container and target port for the ALB configuration.
+func (s *BackendService) HTTPLoadBalancerTarget(exposedPorts []ExposedPort, rrTargetContainer *string, rrTargetPort *uint16) (targetContainer *string, targetPort *string) {
+	// Route load balancer traffic to main container by default.
+	targetContainer = s.Name
+	targetPort = aws.String(s.ContainerPort())
+
+	rrTarget := rrTargetContainer
+	if rrTarget != nil && *rrTarget != *targetContainer {
+		targetContainer = rrTarget
+		targetPort = s.Sidecars[aws.StringValue(targetContainer)].Port
+	}
+	// Route load balancer traffic to the target_port if mentioned.
+	if container, port := httpLoadBalancerTarget(exposedPorts, rrTargetPort); port != nil {
+		targetPort = port
+	} else if container != nil {
+		targetContainer = container
+	}
```

Comment:
The logic seems confusing. I feel like we should

1. if target port and target container both are empty, return the main container/port
2. if target port is empty, return target container and it's only port
3. if target container is empty, find the container and return the container and target port
4. else (both are not empty), return target container/target port

Also both `rrTargetContainer` and `rrTargetPort` are from `manifest.RoutingRule` which we can access from `s`. Why do we make them parameters?

Reply:
I think I agree to the steps that you've mentioned, and that is what the current logic also does. May be to simplify I can follow your steps. 

> Also both rrTargetContainer and rrTargetPort are from manifest.RoutingRule which we can access from s. Why do we make them parameters?

This is a great question - we can not always take `rrTargetContainer` and `rrTargetPort` from `manifest.RoutingRule` because when `additional_rules` will come into picture then targetContainer and targetPort will be from `manifest.RoutingRule.additional_rules` and not from `manifest.RoutingRule`. Does that make sense?